src/data/nat/digits.lean

```diff
@@ -456,6 +460,52 @@ begin
   exact base_pow_length_digits_le' b m,
 end
 
+lemma of_digits_zeros {α : Type*} [semiring α] (b : α) {L : list ℕ} (heq : ∀ a ∈ L, a = 0) :
+  of_digits b L = 0 :=
+begin
+  induction L with d L ih,
+  { exact of_digits_nil b, },
+  { rw [of_digits, heq d (list.mem_cons_self d L), cast_zero, zero_add],
+    refine mul_eq_zero_of_right b (ih _),
+    exact λ a' ah, heq a' (list.mem_cons_of_mem d ah), }
+end
+
+/--
+Applying `(digits b ∘ of_digits b)` will create a prefix of the original list if it has proper
+digits; it removes trailing zeros.
+-/
+lemma digits_of_digits_prefix {b : ℕ} {L : list ℕ} (hlt : ∀ a ∈ L, a < b) :
+  digits b (of_digits b L) <+: L :=
```

Comment:
As stated, it feels like this is not a very useful lemma. Surely we want to know what is actually stated in the doc-string, i.e. that you get exactly `L` with trailing zeros removed?

Given that saying "`L` with trailing zeroes removed" is perhaps slightly cumbersome, why not instead at the hypothesis that `L` does not end with `0`, and then strengthen the conclusion?

A separate lemma could say that `of_digits b (L ++ [0]) = of_digits b L`.

Reply:
(I'm also happy to be convinced that this lemma is useful as is, although I have to say that I'm skeptical that Kaprekar's routine counts as an application. :-)

Reply:
(Also --- thanks for the PR! Sorry if the above comment sounds grumpy, I'm just wondering what lemma you actually want.)